<!doctype html>
<html lang="">
<head>
	<meta charset="utf-8"/>
	<title>phylogenetics - Make More Machines</title>
	<meta name="author" content="Paul Agapow">


	<link rel="stylesheet" href="/theme/css/main.css" type="text/css" />



</head>

<body>

    <div class="container">

	  <header role="banner">
	    <div class="feeds">
	    </div>
	      <nav class="pages">
			  <a href="/about">About</a>
-			  <a href="/whereabouts">Whereabouts</a>
	      </nav>
		<a href="" class="title">Make More Machines</a>&nbsp;&nbsp;(agapow.net)
      </header>

	  <div class="wrapper">

		  <div role="main" class="content">

<h1>Post tagged: phylogenetics</h1>

<article class="summary">
<div class="metadata">
  <time datetime="2012-01-01T12:00:00+00:00" pubdate>2012-01-01</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/bioruby/">Bioruby</a>, <a href="//by-tag/phylogenetics/">phylogenetics</a>, <a href="//by-tag/visualisation/">visualisation</a>, <a href="//by-tag/dendroscope/">Dendroscope</a></p></div>  <div>
    <h2>
      <a href="/science/computational-biology/tools/coloring-dendroscope-files/" rel="bookmark" title="Permalink to Coloring dendroscope files">Coloring dendroscope files</a>
    </h2>
		<div class="summary">How to programatically label phylogenies.</div>
   <p>The need had arisen for the tips of a large phylogeny to be labelled in
a systematic way. Rather than &quot;point and click&quot; within Dendroscope, this
script takes a .den/dendro file and colors the tips according to a
&quot;color description&quot; file. This is a simple csv file with taxa …</p>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2010-03-01T12:00:00+00:00" pubdate>2010-03-01</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/science/">science</a>, <a href="//by-tag/tools/">tools</a>, <a href="//by-tag/phylogenetics/">phylogenetics</a>, <a href="//by-tag/web-development/">web-development</a></p></div>  <div>
    <h2>
      <a href="/science/computational-biology/jsphylosvg/" rel="bookmark" title="Permalink to jsPhyloSVG">jsPhyloSVG</a>
    </h2>
   <p><a class="reference external" href="http://jsphylosvg.com/">jsPhyloSVG</a> (or JPS from this point) is a
nifty Javascript library for displaying phylogenetic trees in your
browser. It can:</p>
<ul class="simple">
<li>accept trees in a number of formats, encoded in the page or on file or loaded via Ajax</li>
<li>allow these trees to be exported or saved as SVG</li>
<li>display trees …</li></ul>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2009-01-01T12:00:00+00:00" pubdate>2009-01-01</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/mr-bayes/">mr-bayes</a>, <a href="//by-tag/phylogenetics/">phylogenetics</a>, <a href="//by-tag/possibly-obselete/">possibly-obselete</a></p>     <div class="attention"><P><b>Attention:</b> This article may refer to information that is outdated or no longer relevant. It is left here for historical purposes.</P></div>
</div>  <div>
    <h2>
      <a href="/science/computational-biology/tools/error-1-for-mrbayes/" rel="bookmark" title="Permalink to Error 1 for Mrbayes">Error 1 for Mrbayes</a>
    </h2>
		<div class="summary">What happens when a make fails.</div>
   <p>If this happens to you when trying to compile MrBayes:</p>
<pre class="literal-block">
% make
gcc -DUNIX_VERSION -DUSE_READLINE -O3 -Wall    -c -o mb.o mb.c
gcc -DUNIX_VERSION -DUSE_READLINE -O3 -Wall    -c -o mcmc.o mcmc.c
gcc -DUNIX_VERSION -DUSE_READLINE -O3 -Wall    -c -o bayes.o bayes.c
bayes.c:45:31: readline/readline …</pre>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2006-06-01T12:00:00+01:00" pubdate>2006-06-01</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/software/">software</a>
<p class="tags">tagged <a href="//by-tag/software/">software</a>, <a href="//by-tag/python/">python</a>, <a href="//by-tag/conservation/">conservation</a>, <a href="//by-tag/taxonomy/">taxonomy</a>, <a href="//by-tag/phylogenetics/">phylogenetics</a>, <a href="//by-tag/biodiversity/">biodiversity</a></p></div>  <div>
    <h2>
      <a href="/software/treemaker/" rel="bookmark" title="Permalink to TreeMaker">TreeMaker</a>
    </h2>
		<div class="summary">An application for the interactive building of taxonomies.</div>
   <div class="section" id="background">
<h2>Background</h2>
<p>Biodiversity assessment demands objective measures, because ultimately
conservation is an issue of economics, prioritizing the use of limited
resources for preserving taxa. The most general framework for such
metrics are those that assess evolutionary distinctiveness as judged by
how much of a phylogeny is conserved. However, their applicability is …</p></div>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2004-03-01T12:00:00+00:00" pubdate>2004-03-01</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/software/">software</a>
<p class="tags">tagged <a href="//by-tag/phylogenetics/">phylogenetics</a></p></div>  <div>
    <h2>
      <a href="/software/mac5/" rel="bookmark" title="Permalink to Mac5">Mac5</a>
    </h2>
		<div class="summary">A program for phylogenetic reconstruction using gapped data.</div>
   <div class="section" id="background">
<h2>Background</h2>
<p>MAC5 is a program which implements MCMC sampling to estimate a
phylogenetic tree from a DNA multiple alignment. What differentiates
MAC5 from similar programs (e.g. BAMBE, MrBayes) is its use of
five-state sequence evolution models as a means to include the gap
information when estimating an alignment. Despite …</p></div>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2003-06-01T12:00:00+01:00" pubdate>2003-06-01</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/software/">software</a>
<p class="tags">tagged <a href="//by-tag/software/">software</a>, <a href="//by-tag/python/">python</a>, <a href="//by-tag/phylogenetics/">phylogenetics</a>, <a href="//by-tag/macroevolution/">macroevolution</a>, <a href="//by-tag/biodiversity/">biodiversity</a></p></div>  <div>
    <h2>
      <a href="/software/mesa/" rel="bookmark" title="Permalink to MeSA">MeSA</a>
    </h2>
		<div class="summary">Macroevolutionary Analysis & Simulation.</div>
   <div class="section" id="background">
<h2>Background</h2>
<p>Phylogenies are smoking guns. An evolutionary history carrys in its very
shape the fingerprints of every process that effected its creation:
extinctions, key adaptations, radiations, colonizations, changes in the
tempo and mode of macroevolution.</p>
<p>Unfortunately analysing phylogenies is difficult. The calculations of
interest to any investigator are sometimes intricate …</p></div>
  </div>
</article>

		  </div>

		  <div class="sidebar">
		    <div class="sidebar-container" >


	            <aside>
	              <h2>About</h2>
			      <p class='summary'>
                  This site has just been restored, with most of the content
						translated automatically from its old form. Expect bugs,
						missing information and general wierdness as things
						get sorted out.
			      </p>
			    </aside>


  	          <nav>
	            <h2>Categories</h2>
	            <ul>
	                <li ><a href="/misc/">misc</a></li>
	                <li ><a href="/personal/">personal</a></li>
	                <li ><a href="/programming/">programming</a></li>
	                <li ><a href="/publications/">publications</a></li>
	                <li ><a href="/science/">science</a></li>
	                <li ><a href="/software/">software</a></li>
	                <li ><a href="/talks/">talks</a></li>
	            </ul>

	            <ul>
	              <li><a href="/tags.html">all tags</a></li>
	            </ul>
	          </nav>

	            <aside>
	            <h2>Social</h2>
			      <ul class="social">
				    <li><a href="http://twitter.com/agapow">Twitter</a><i></i></li>
				    <li><a href="http://github.com/agapow">Github</a><i></i></li>
				    <li><a href="https://plus.google.com/u/0/113920832845886210876/posts">Google+</a><i></i></li>
				    <li><a href="http://uk.linkedin.com/pub/paul-michael-agapow/2/15b/184/">linkedIn</a><i></i></li>
			      </ul>
			    </aside>

	            <aside>
	              <h2>Links</h2>
	              <ul>
	                  <li><a href="http://www.imperial.ac.uk/data-science/">IC Data Science Institute</a></li>
	                  <li><a href="http://www.meetup.com/Bioinformatics-London">Bioinformatics London</a></li>
	              </ul>
	            </aside>

<a class="twitter-timeline" data-dnt="true" href="https://twitter.com/agapow" data-widget-id="336428520026812416">Tweets by @agapow</a>
<script>!function(d,s,id){var js,fjs=d.getElementsByTagName(s)[0],p=/^http:/.test(d.location)?'http':'https';if(!d.getElementById(id)){js=d.createElement(s);js.id=id;js.src=p+"://platform.twitter.com/widgets.js";fjs.parentNode.insertBefore(js,fjs);}}(document,"script","twitter-wjs");</script>

	        </div>
		  </div>

	  </div>

      <footer>
		<p role="contentinfo">
		  Paul Agapow - Proudly powered by <a href="http://alexis.notmyidea.org/pelican/">pelican</a>. Theme <a href="https://github.com/fle/pelican-sober">pelican-sober</a>.
    	</p>

	  </footer>

	</div>


</body>
</html>