<!doctype html>
<html lang="">
<head>
	<meta charset="utf-8"/>
	<title> - Make More Machines</title>
	<meta name="author" content="Paul Agapow">


	<link rel="stylesheet" href="/theme/css/main.css" type="text/css" />



</head>

<body>

    <div class="container">

	  <header role="banner">
	    <div class="feeds">
	    </div>
	      <nav class="pages">
			  <a href="/about">About</a>
-			  <a href="/whereabouts">Whereabouts</a>
	      </nav>
		<a href="" class="title">Make More Machines</a>&nbsp;&nbsp;(agapow.net)
      </header>

	  <div class="wrapper">

		  <div role="main" class="content">

<h1>Posts categorized under: </h1>

<article class="summary">
<div class="metadata">
  <time datetime="2012-01-01T12:00:00+00:00" pubdate>2012-01-01</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
</div>  <div>
    <h2>
      <a href="/science/computational-biology/scripts/cleaning-biosequences/" rel="bookmark" title="Permalink to Cleaning biosequences">Cleaning biosequences</a>
    </h2>
   <p><em>A simple script to check and purge sequence files of possible
problems.</em></p>
<p>Some times you need sequences that are unambiguous (i.e. only 'ACGT',
lacking gaps) whether it's because of the limitations or assumptions of
tools (like omegaMap) or just because you want to know where SNPs or
sequencing ambiguities …</p>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2011-01-01T12:00:00+00:00" pubdate>2011-01-01</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/ruby/">ruby</a>, <a href="//by-tag/bioinformatics/">bioinformatics</a>, <a href="//by-tag/sequence-analysis/">sequence-analysis</a></p></div>  <div>
    <h2>
      <a href="/science/computational-biology/scripts/drawing-sequence-logos/" rel="bookmark" title="Permalink to Drawing sequence logos">Drawing sequence logos</a>
    </h2>
		<div class="summary">A very simple script to do a simple but tedious task.</div>
   <p>Sequence logos are a common way of representing SNPs and diversity in
groups of sequences. This script automates the task. It's a bit rough
around the edges and serves mainly as a base for further hacking.</p>
<p>Usage is:</p>
<pre class="literal-block">
drawlogo.rb [options] FILE1 [FILE2 ...]
</pre>
<p>where options are:</p>
<blockquote>
<table class="docutils option-list" frame="void" rules="none">
<col class="option" />
<col class="description" />
<tbody valign="top">
<tr><td class="option-group">
<kbd><span class="option">-h</span>, <span class="option">--help</span></kbd></td>
<td>Display this …</td></tr></tbody></table></blockquote>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2010-06-01T12:00:00+01:00" pubdate>2010-06-01</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/imported/">imported</a>, <a href="//by-tag/sequence-analysis/">sequence-analysis</a>, <a href="//by-tag/script/">script</a>, <a href="//by-tag/ruby/">ruby</a>, <a href="//by-tag/bioruby/">bioruby</a>, <a href="//by-tag/bioinformatics/">bioinformatics</a></p>     <div class="attention"><b>Attention:</b> This article has been imported from a previous website and has not yet been checked. It may be malformed or incomplete.
     </div>
</div>  <div>
    <h2>
      <a href="/science/computational-biology/scripts/reducing-a-sequence-to-snps/" rel="bookmark" title="Permalink to Reducing a sequence to SNPs">Reducing a sequence to SNPs</a>
    </h2>
		<div class="summary">A script for a simple task.</div>
   <p>A largely self-explanatory script. This will &quot;shrink&quot; an alignment,
deleting all sites that don't contain a polymorphism in some member
sequence. A little bit of script candy as well, this takes any number of
files and saves the results in a new file named according to a definable
schema:</p>
<pre class="literal-block">
#!/usr …</pre>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2007-06-14T12:56:00+01:00" pubdate>2007-06-14</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/bioruby/">bioruby</a>, <a href="//by-tag/sequences/">sequences</a>, <a href="//by-tag/database/">database</a></p></div>  <div>
    <h2>
      <a href="/science/computational-biology/scripts/fetch-sequences-from-db/" rel="bookmark" title="Permalink to Fetch sequences from db">Fetch sequences from db</a>
    </h2>
		<div class="summary">A simple script to grab bioseqs by accession.</div>
   <p>This just wraps the BioRuby fetch functionality in a friendly
commandline interface. In brief, it can accept accession ids on the
commandline or from a piped file (one accession per line) and save the
corresponding sequences from the db. Sequences may be downloaded via the
bioruby or EBI servers. The …</p>
  </div>
</article>

		  </div>

		  <div class="sidebar">
		    <div class="sidebar-container" >


	            <aside>
	              <h2>About</h2>
			      <p class='summary'>
                  This site has just been restored, with most of the content
						translated automatically from its old form. Expect bugs,
						missing information and general wierdness as things
						get sorted out.
			      </p>
			    </aside>


  	          <nav>
	            <h2>Categories</h2>
	            <ul>
	                <li ><a href="/misc/">misc</a></li>
	                <li ><a href="/personal/">personal</a></li>
	                <li ><a href="/programming/">programming</a></li>
	                <li ><a href="/publications/">publications</a></li>
	                <li ><a href="/science/">science</a></li>
	                <li ><a href="/software/">software</a></li>
	                <li ><a href="/talks/">talks</a></li>
	            </ul>

	            <ul>
	              <li><a href="/tags.html">all tags</a></li>
	            </ul>
	          </nav>

	            <aside>
	            <h2>Social</h2>
			      <ul class="social">
				    <li><a href="http://twitter.com/agapow">Twitter</a><i></i></li>
				    <li><a href="http://github.com/agapow">Github</a><i></i></li>
				    <li><a href="https://plus.google.com/u/0/113920832845886210876/posts">Google+</a><i></i></li>
				    <li><a href="http://uk.linkedin.com/pub/paul-michael-agapow/2/15b/184/">linkedIn</a><i></i></li>
			      </ul>
			    </aside>

	            <aside>
	              <h2>Links</h2>
	              <ul>
	                  <li><a href="http://www.imperial.ac.uk/data-science/">IC Data Science Institute</a></li>
	                  <li><a href="http://www.meetup.com/Bioinformatics-London">Bioinformatics London</a></li>
	              </ul>
	            </aside>

<a class="twitter-timeline" data-dnt="true" href="https://twitter.com/agapow" data-widget-id="336428520026812416">Tweets by @agapow</a>
<script>!function(d,s,id){var js,fjs=d.getElementsByTagName(s)[0],p=/^http:/.test(d.location)?'http':'https';if(!d.getElementById(id)){js=d.createElement(s);js.id=id;js.src=p+"://platform.twitter.com/widgets.js";fjs.parentNode.insertBefore(js,fjs);}}(document,"script","twitter-wjs");</script>

	        </div>
		  </div>

	  </div>

      <footer>
		<p role="contentinfo">
		  Paul Agapow - Proudly powered by <a href="http://alexis.notmyidea.org/pelican/">pelican</a>. Theme <a href="https://github.com/fle/pelican-sober">pelican-sober</a>.
    	</p>

	  </footer>

	</div>


</body>
</html>