<!doctype html>
<html lang="">
<head>
	<meta charset="utf-8"/>
	<title> - Make More Machines</title>
	<meta name="author" content="Paul Agapow">


	<link rel="stylesheet" href="/theme/css/main.css" type="text/css" />



</head>

<body>

    <div class="container">

	  <header role="banner">
	    <div class="feeds">
	    </div>
	      <nav class="pages">
			  <a href="/about">About</a>
-			  <a href="/whereabouts">Whereabouts</a>
	      </nav>
		<a href="/" class="title">Make More Machines</a>&nbsp;&nbsp;(agapow.net)
      </header>

	  <div class="wrapper">

		  <div role="main" class="content">

<h1>Posts categorized under: </h1>

<article class="summary">
<div class="metadata">
  <time datetime="2017-06-11T12:00:00+01:00" pubdate>2017-06-11</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/bioinformatics/">bioinformatics</a>, <a href="//by-tag/academia/">academia</a></p></div>  <div>
    <h2>
      <a href="/science/computational-biology/no-respect/" rel="bookmark" title="Permalink to Why bioinformaticians don't get no respect">Why bioinformaticians don't get no respect</a>
    </h2>
   <p>It's a common cry amongst working bioinformaticians that they're unappreciated, undervalued and generally &quot;get no respect&quot;. While people love to complain, from discussions with peers and colleagues, the same stories come up again and again:</p>
<ul class="simple">
<li>Not being consulted when projects are planned</li>
<li>Technical advice and results not taken seriously</li>
<li>Being …</li></ul>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2017-05-01T12:00:00+01:00" pubdate>2017-05-01</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/programming/">programming</a>, <a href="//by-tag/software/">software</a>, <a href="//by-tag/research/">research</a>, <a href="//by-tag/amazon/">amazon</a>, <a href="//by-tag/aws/">aws</a></p></div>  <div>
    <h2>
      <a href="/science/computational-biology/aws-for-research/" rel="bookmark" title="Permalink to Using AWS for research computing">Using AWS for research computing</a>
    </h2>
		<div class="summary">Your own private computing cluster?</div>
   <p>This is based upon my reply to a question on reddit concerning experiences with using Amazon Web Services (including ELastic Computing, Glacier, etc.) for research.</p>
<p>I was part of a pan-European research consortium that used it for our shared computing infrastructure: databases, a few web apps and web sites, mailers …</p>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2015-02-25T23:29:00+00:00" pubdate>2015-02-25</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/bioinformatics/">bioinformatics</a>, <a href="//by-tag/career/">career</a>, <a href="//by-tag/humour/">humour</a></p></div>  <div>
    <h2>
      <a href="/science/computational-biology/words-not-to-hear/" rel="bookmark" title="Permalink to Words a bioinformatician never wants to hear">Words a bioinformatician never wants to hear</a>
    </h2>
		<div class="summary">Based on hard-won experience.</div>
   <p>(This first appeared on biocodershub.net courtesy of Rad, and has since popped up on coderscrowd.com. It enjoyed some moments of viral popularity, with many aggrieved practitioners chipping in on the comments of the article. Following the resurrection of my website, it's a good opportunity to bring this piece …</p>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2013-07-11T13:30:00+01:00" pubdate>2013-07-11</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/biopython/">BioPython</a>, <a href="//by-tag/biosequences/">biosequences</a>, <a href="//by-tag/computational-biology/">computational biology</a>, <a href="//by-tag/python/">Python</a></p></div>  <div>
    <h2>
      <a href="/science/computational-biology/biofoo/hitchhikers-guide-to-biopython-sequences-alphabets/" rel="bookmark" title="Permalink to Hitchhikers guide to Biopython: Sequences & alphabets">Hitchhikers guide to Biopython: Sequences &amp; alphabets</a>
    </h2>
   <p>(Originally published on BiocodersHub.)</p>
<p>If you’re doing bioinformatics in Python, you’re probably
using  <a class="reference external" href="http://biopython.org/">Biopython</a>. Actually, Biopython is a
good reason for using Python. But it can be formidable to newcomers:
there’s a lot there and there’s not a huge amount of learning material.
This then is …</p>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2012-06-14T12:56:00+01:00" pubdate>2012-06-14</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/computational-biology/">computational biology</a>, <a href="//by-tag/galaxy/">galaxy</a></p></div>  <div>
    <h2>
      <a href="/science/computational-biology/galaxy/common-tasks-in-galaxy/" rel="bookmark" title="Permalink to Common tasks in Galaxy">Common tasks in Galaxy</a>
    </h2>
   <p><em>It's all there in the documentation, but sometimes it's hard to find.
This document gives you another place to look.</em></p>
<p>So how do I ...</p>
<div class="section" id="create-admin-users">
<h2>... create admin users?</h2>
<p>Curiously, the identity of admin users is hardcoded into the Galaxy
configuration file. (Which makes it secure, I guess, but separate from
the …</p></div>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2012-06-14T12:56:00+01:00" pubdate>2012-06-14</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
</div>  <div>
    <h2>
      <a href="/science/computational-biology/tools/compiling-quickjoin-and-file-formats/" rel="bookmark" title="Permalink to Compiling Quickjoin and file formats">Compiling Quickjoin and file formats</a>
    </h2>
   <p><em>Problems with building qjoin and getting it to read stockholm files.</em></p>
<p><a class="reference external" href="http://www.birc.dk/Software/QuickJoin">Quickjoin / qjoin</a> is an
excellent commandline program for rapid construction of
neighbour-joining trees. However, while using it recently, I had a few
problems getting it to read Stockholm files, the most accessible of the
formats it can use.</p>
<p>The …</p>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2012-06-14T12:56:00+01:00" pubdate>2012-06-14</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/computational-biology/">computational biology</a>, <a href="//by-tag/galaxy/">galaxy</a></p></div>  <div>
    <h2>
      <a href="/science/computational-biology/galaxy/galaxy-toolsheds/" rel="bookmark" title="Permalink to Galaxy toolsheds">Galaxy toolsheds</a>
    </h2>
   <p>Galaxy toolsheds</p>
<p>Relatively painless tool-sharing</p>
<p>This is a more recent innovation in Galaxy, which can make it a
somewhat confused one: the concept of the toolshed has changed over its
lifetime, the documentation is incomplete, and there's a slightly
strange emphasis in the documentation that exists. So …</p>
<div class="section" id="mile-high-description">
<h2>Mile-high description</h2>
<p>Toolsheds …</p></div>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2012-06-01T12:00:00+01:00" pubdate>2012-06-01</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/computational-biology/">computational-biology</a>, <a href="//by-tag/programming/">programming</a>, <a href="//by-tag/programming-langauges/">programming-langauges</a>, <a href="//by-tag/python/">python</a>, <a href="//by-tag/ruby/">ruby</a></p></div>  <div>
    <h2>
      <a href="/science/computational-biology/language-wars/" rel="bookmark" title="Permalink to Language Wars">Language Wars</a>
    </h2>
		<div class="summary">"What language would you recommend to introduce programming to an audience of life science students at a bachelor level?"</div>
   <p>(Originally published on BiocodersHub)</p>
<p>Following several lengthy and passionate discussions in different venues
on what language to use for teaching bioinformatics, I've started
cutting and pasting my reply. And here it is.</p>
<hr class="docutils" />
<p>You'll get a lot of different opinions on this because:</p>
<ul class="simple">
<li>It's a religious issue. That is, it comes …</li></ul>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2012-02-01T12:00:00+00:00" pubdate>2012-02-01</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/biopython/">BioPython</a>, <a href="//by-tag/biosequences/">biosequences</a>, <a href="//by-tag/computational-biology/">computational biology</a>, <a href="//by-tag/programming/">programming</a>, <a href="//by-tag/python/">Python</a></p></div>  <div>
    <h2>
      <a href="/science/computational-biology/biofoo/hitchhikers-guide-to-biopython-seqrecords/" rel="bookmark" title="Permalink to Hitchhikers guide to BioPython: SeqRecords">Hitchhikers guide to BioPython: SeqRecords</a>
    </h2>
		<div class="summary">For the novice, more-than-raw sequences.</div>
   <p>(Previously published on BiocodersHub.)</p>
<p>Previously I'd spoken about how Biopython represents sequence data with
the Seq class. But there is also the SeqRecord class:</p>
<ul class="simple">
<li>A Seq is just raw sequence data and information about what type of sequence it is.</li>
<li>A SeqRecord is a Seq and all the other information …</li></ul>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2012-01-01T12:00:00+00:00" pubdate>2012-01-01</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
</div>  <div>
    <h2>
      <a href="/science/computational-biology/scripts/cleaning-biosequences/" rel="bookmark" title="Permalink to Cleaning biosequences">Cleaning biosequences</a>
    </h2>
   <p><em>A simple script to check and purge sequence files of possible
problems.</em></p>
<p>Some times you need sequences that are unambiguous (i.e. only 'ACGT',
lacking gaps) whether it's because of the limitations or assumptions of
tools (like omegaMap) or just because you want to know where SNPs or
sequencing ambiguities …</p>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2012-01-01T12:00:00+00:00" pubdate>2012-01-01</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/bioruby/">Bioruby</a>, <a href="//by-tag/phylogenetics/">phylogenetics</a>, <a href="//by-tag/visualisation/">visualisation</a>, <a href="//by-tag/dendroscope/">Dendroscope</a></p></div>  <div>
    <h2>
      <a href="/science/computational-biology/tools/coloring-dendroscope-files/" rel="bookmark" title="Permalink to Coloring dendroscope files">Coloring dendroscope files</a>
    </h2>
		<div class="summary">How to programatically label phylogenies.</div>
   <p>The need had arisen for the tips of a large phylogeny to be labelled in
a systematic way. Rather than &quot;point and click&quot; within Dendroscope, this
script takes a .den/dendro file and colors the tips according to a
&quot;color description&quot; file. This is a simple csv file with taxa …</p>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2012-01-01T12:00:00+00:00" pubdate>2012-01-01</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/bioinformatics/">bioinformatics</a>, <a href="//by-tag/bioruby/">BioRuby</a>, <a href="//by-tag/sequence-analysis/">sequence analysis</a></p></div>  <div>
    <h2>
      <a href="/science/computational-biology/biofoo/consensus-in-bioruby/" rel="bookmark" title="Permalink to Consensus in BioRuby">Consensus in BioRuby</a>
    </h2>
		<div class="summary">Explaining the ill-explained ways to obtain a consensus sequence in BioRuby.</div>
   <p>In BioRuby, alignments are equipped with several methods for obtaining
consensus sequences. Unfortunately, these have terse descriptions which
point you at the BioPerl documentation, with the added bonus of not
<em>quite</em> working like the BioPerl equivalents.</p>
<p>First, let's create a very simple alignment, where everything agrees
except the last sequence …</p>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2012-01-01T12:00:00+00:00" pubdate>2012-01-01</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/computational-biology/">computational biology</a>, <a href="//by-tag/galaxy/">galaxy</a></p></div>  <div>
    <h2>
      <a href="/science/computational-biology/galaxy/galaxy-miscellanea/" rel="bookmark" title="Permalink to Galaxy miscellanea">Galaxy miscellanea</a>
    </h2>
   <p><em>Odds and ends and the surprising.</em></p>
<div class="section" id="redirects">
<h2>Redirects</h2>
<p>If you are serving the installation with a proxy redirect (e.g. the
galaxy server is running on port 7070 but is being redirect by Apache to
appear at port 80 on <em>/galaxy</em>), while you can access Galaxy at both
addresses, login will …</p></div>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2012-01-01T12:00:00+00:00" pubdate>2012-01-01</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
</div>  <div>
    <h2>
      <a href="/science/computational-biology/tools/more-about-mrbayes/" rel="bookmark" title="Permalink to More about MrBayes">More about MrBayes</a>
    </h2>
   <p>Some (more) notes about the venerable Bayesian reconstruction program.</p>
<div class="section" id="error-when-setting-parameter-gap-2">
<h2>Error when setting parameter &quot;Gap&quot; (2)</h2>
<p>When attempting to execute a Nexus file, MrBayes kept spitting back this
cryptic error upon loading:</p>
<pre class="literal-block">
Executing file &quot;c_vp1_nuc_seqs.nxs&quot; [...]
Reading data block Allocated matrix [...]
Data is Dna Gap character matches matching or missing characters …</pre></div>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2012-01-01T12:00:00+00:00" pubdate>2012-01-01</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/computational-biology/">computational biology</a>, <a href="//by-tag/galaxy/">galaxy</a></p></div>  <div>
    <h2>
      <a href="/science/computational-biology/galaxy/more-things-i-done-learned-about-galaxy-tool-development/" rel="bookmark" title="Permalink to More things I done learned about Galaxy tool development">More things I done learned about Galaxy tool development</a>
    </h2>
   <p><em>More things I done learned about Galaxy tool development</em></p>
<p>A pot-pouri of titbits that are probably documented somewhere, but
weren't obvious to me.</p>
<div class="section" id="labels-on-the-tool-menu">
<h2>Labels on the tool menu</h2>
<p>You can't nest sections on the tool panel - but you can put labels in
sections for the same effect. Look at tool_conf …</p></div>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2012-01-01T12:00:00+00:00" pubdate>2012-01-01</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/science/">science</a>, <a href="//by-tag/bioinformatics/">bioinformatics</a>, <a href="//by-tag/ngs/">ngs</a></p></div>  <div>
    <h2>
      <a href="/science/computational-biology/what-works-ngs-assemblers/" rel="bookmark" title="Permalink to What works - NGS assemblers">What works - NGS assemblers</a>
    </h2>
		<div class="summary">A quick paper review on picking the best assembler</div>
   <p>You could spend all day just keeping up with developments in
next-generation sequencing. Companies announce new and revolutionary
technologies seemingly every month, promising to do more, better and for
less. Yet at the same time, it’s difficult to hack your way through the
marketing tallk and get hard figures …</p>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2011-01-01T12:00:00+00:00" pubdate>2011-01-01</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/ruby/">ruby</a>, <a href="//by-tag/bioinformatics/">bioinformatics</a>, <a href="//by-tag/sequence-analysis/">sequence-analysis</a></p></div>  <div>
    <h2>
      <a href="/science/computational-biology/scripts/drawing-sequence-logos/" rel="bookmark" title="Permalink to Drawing sequence logos">Drawing sequence logos</a>
    </h2>
		<div class="summary">A very simple script to do a simple but tedious task.</div>
   <p>Sequence logos are a common way of representing SNPs and diversity in
groups of sequences. This script automates the task. It's a bit rough
around the edges and serves mainly as a base for further hacking.</p>
<p>Usage is:</p>
<pre class="literal-block">
drawlogo.rb [options] FILE1 [FILE2 ...]
</pre>
<p>where options are:</p>
<blockquote>
<table class="docutils option-list" frame="void" rules="none">
<col class="option" />
<col class="description" />
<tbody valign="top">
<tr><td class="option-group">
<kbd><span class="option">-h</span>, <span class="option">--help</span></kbd></td>
<td>Display this …</td></tr></tbody></table></blockquote>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2010-06-01T12:00:00+01:00" pubdate>2010-06-01</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/imported/">imported</a>, <a href="//by-tag/sequence-analysis/">sequence-analysis</a>, <a href="//by-tag/script/">script</a>, <a href="//by-tag/ruby/">ruby</a>, <a href="//by-tag/bioruby/">bioruby</a>, <a href="//by-tag/bioinformatics/">bioinformatics</a></p>     <div class="attention"><b>Attention:</b> This article has been imported from a previous website and has not yet been checked. It may be malformed or incomplete.
     </div>
</div>  <div>
    <h2>
      <a href="/science/computational-biology/scripts/reducing-a-sequence-to-snps/" rel="bookmark" title="Permalink to Reducing a sequence to SNPs">Reducing a sequence to SNPs</a>
    </h2>
		<div class="summary">A script for a simple task.</div>
   <p>A largely self-explanatory script. This will &quot;shrink&quot; an alignment,
deleting all sites that don't contain a polymorphism in some member
sequence. A little bit of script candy as well, this takes any number of
files and saves the results in a new file named according to a definable
schema:</p>
<pre class="literal-block">
#!/usr …</pre>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2010-03-01T12:00:00+00:00" pubdate>2010-03-01</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/science/">science</a>, <a href="//by-tag/tools/">tools</a>, <a href="//by-tag/phylogenetics/">phylogenetics</a>, <a href="//by-tag/web-development/">web-development</a></p></div>  <div>
    <h2>
      <a href="/science/computational-biology/jsphylosvg/" rel="bookmark" title="Permalink to jsPhyloSVG">jsPhyloSVG</a>
    </h2>
   <p><a class="reference external" href="http://jsphylosvg.com/">jsPhyloSVG</a> (or JPS from this point) is a
nifty Javascript library for displaying phylogenetic trees in your
browser. It can:</p>
<ul class="simple">
<li>accept trees in a number of formats, encoded in the page or on file or loaded via Ajax</li>
<li>allow these trees to be exported or saved as SVG</li>
<li>display trees …</li></ul>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2009-01-01T12:00:00+00:00" pubdate>2009-01-01</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/mr-bayes/">mr-bayes</a>, <a href="//by-tag/phylogenetics/">phylogenetics</a>, <a href="//by-tag/possibly-obselete/">possibly-obselete</a></p>     <div class="attention"><P><b>Attention:</b> This article may refer to information that is outdated or no longer relevant. It is left here for historical purposes.</P></div>
</div>  <div>
    <h2>
      <a href="/science/computational-biology/tools/error-1-for-mrbayes/" rel="bookmark" title="Permalink to Error 1 for Mrbayes">Error 1 for Mrbayes</a>
    </h2>
		<div class="summary">What happens when a make fails.</div>
   <p>If this happens to you when trying to compile MrBayes:</p>
<pre class="literal-block">
% make
gcc -DUNIX_VERSION -DUSE_READLINE -O3 -Wall    -c -o mb.o mb.c
gcc -DUNIX_VERSION -DUSE_READLINE -O3 -Wall    -c -o mcmc.o mcmc.c
gcc -DUNIX_VERSION -DUSE_READLINE -O3 -Wall    -c -o bayes.o bayes.c
bayes.c:45:31: readline/readline …</pre>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2007-06-14T12:56:00+01:00" pubdate>2007-06-14</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/bioruby/">bioruby</a>, <a href="//by-tag/sequences/">sequences</a>, <a href="//by-tag/database/">database</a></p></div>  <div>
    <h2>
      <a href="/science/computational-biology/scripts/fetch-sequences-from-db/" rel="bookmark" title="Permalink to Fetch sequences from db">Fetch sequences from db</a>
    </h2>
		<div class="summary">A simple script to grab bioseqs by accession.</div>
   <p>This just wraps the BioRuby fetch functionality in a friendly
commandline interface. In brief, it can accept accession ids on the
commandline or from a piped file (one accession per line) and save the
corresponding sequences from the db. Sequences may be downloaded via the
bioruby or EBI servers. The …</p>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2007-06-14T12:56:00+01:00" pubdate>2007-06-14</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/computational-biology/">computational biology</a>, <a href="//by-tag/galaxy/">galaxy</a></p></div>  <div>
    <h2>
      <a href="/science/computational-biology/galaxy/installing-galaxy/" rel="bookmark" title="Permalink to Installing Galaxy">Installing Galaxy</a>
    </h2>
		<div class="summary">Setting up a production version of GMOD Galaxy for general use.</div>
   <p>This presents one way to create an optimized production Galaxy instance.
Variations are certainly possible and some of the choices presented
are/were dictated by local culture. Certain settings may be more
suitable for production or development environments. Nonetheless, this
presents a start-to-stop process for installation and setup.</p>
<p><strong>Note:</strong> this …</p>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2007-01-01T12:00:00+00:00" pubdate>2007-01-01</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
</div>  <div>
    <h2>
      <a href="/science/computational-biology/tools/parsing-dendroscope-nodes/" rel="bookmark" title="Permalink to Parsing Dendroscope nodes">Parsing Dendroscope nodes</a>
    </h2>
   <p>For when you have to do lots to a big tree.</p>
<p>Previously, I showed how Dendroscope files can be easily manipulated
with brute-force regex, so you can right scripts to color a mass of
nodes, rather than having to format them one-by-one in the GUI. However,
more complex manipulations require …</p>
  </div>
</article>

		  </div>

		  <div class="sidebar">
		    <div class="sidebar-container" >


	            <aside>
	              <h2>About</h2>
			      <p class='summary'>
                  This site has just been restored, with most of the content
						translated automatically from its old form. Expect bugs,
						missing information and general wierdness as things
						get sorted out.
			      </p>
			    </aside>


  	          <nav>
	            <h2>Categories</h2>
	            <ul>
	                <li ><a href="/downloads/">downloads</a></li>
	                <li ><a href="/misc/">misc</a></li>
	                <li ><a href="/personal/">personal</a></li>
	                <li ><a href="/programming/">programming</a></li>
	                <li ><a href="/publications/">publications</a></li>
	                <li ><a href="/science/">science</a></li>
	                <li ><a href="/software/">software</a></li>
	                <li ><a href="/talks/">talks</a></li>
	            </ul>

	            <ul>
	              <li><a href="/tags.html">all tags</a></li>
	            </ul>
	          </nav>

	            <aside>
	            <h2>Social</h2>
			      <ul class="social">
				    <li><a href="http://twitter.com/agapow">Twitter</a><i></i></li>
				    <li><a href="http://github.com/agapow">Github</a><i></i></li>
				    <li><a href="https://www.linkedin.com/in/paul-agapow/">LinkedIn</a><i></i></li>
			      </ul>
			    </aside>

	            <aside>
	              <h2>Links</h2>
	              <ul>
	                  <li><a href="http://www.meetup.com/Bioinformatics-London">Bioinformatics London</a></li>
	              </ul>
	            </aside>

<a class="twitter-timeline" data-dnt="true" href="https://twitter.com/agapow" data-widget-id="336428520026812416">Tweets by @agapow</a>
<script>!function(d,s,id){var js,fjs=d.getElementsByTagName(s)[0],p=/^http:/.test(d.location)?'http':'https';if(!d.getElementById(id)){js=d.createElement(s);js.id=id;js.src=p+"://platform.twitter.com/widgets.js";fjs.parentNode.insertBefore(js,fjs);}}(document,"script","twitter-wjs");</script>

	        </div>
		  </div>

	  </div>

      <footer>
		<p role="contentinfo">
		  Paul Agapow - Proudly powered by <a href="http://alexis.notmyidea.org/pelican/">pelican</a>. Theme <a href="https://github.com/fle/pelican-sober">pelican-sober</a>.
    	</p>

	  </footer>

	</div>


</body>
</html>