<!doctype html>
<html lang="">
<head>
	<meta charset="utf-8"/>
	<title>science - Make More Machines</title>
	<meta name="author" content="Paul Agapow">


	<link rel="stylesheet" href="/theme/css/main.css" type="text/css" />



</head>

<body>

    <div class="container">

	  <header role="banner">
	    <div class="feeds">
	    </div>
	      <nav class="pages">
			  <a href="/about">About</a>
-			  <a href="/whereabouts">Whereabouts</a>
	      </nav>
		<a href="" class="title">Make More Machines</a>&nbsp;&nbsp;(agapow.net)
      </header>

	  <div class="wrapper">

		  <div role="main" class="content">

<h1>Posts categorized under: science</h1>

<article class="summary">
<div class="metadata">
  <time datetime="2018-11-19T12:00:00+00:00" pubdate>2018-11-19</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/science/">science</a>, <a href="//by-tag/academia/">academia</a>, <a href="//by-tag/career/">career</a></p></div>  <div>
    <h2>
      <a href="/science/academia/have-you-looked-on-evoldir/" rel="bookmark" title="Permalink to Have you looked on Evoldir?">Have you looked on Evoldir?</a>
    </h2>
		<div class="summary">Reflections on the academic job search</div>
   <p>For those who don’t know, EvolDir is a worldwide mailing list on evolutionary biology that has been running since approximately forever. Everyone who works even vaguely in the area of evolution subscribes to it. Every day It typically carries several posts on conferences, book announcements, funding opportunities and job …</p>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2018-02-20T00:00:00+00:00" pubdate>2018-02-20</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/biology/">biology</a></p></div>  <div>
    <h2>
      <a href="/science/misc/tentacles/" rel="bookmark" title="Permalink to Tentacles vs arms">Tentacles vs arms</a>
    </h2>
		<div class="summary">Solving (partly) an old conundrum</div>
   <p>You might have come across confusing statements like &quot;octopuses have arms, squids have tentacles&quot; and wondered what's the difference. I did and here's the (unsatisfactory) answer.</p>
<p>Cephalopods (squid, octopus, nautilus and a number of other aquatic creatures, which are a class of mollusc) have a number of muscular &quot;limbs&quot;. Traditionally …</p>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2017-07-13T00:00:00+01:00" pubdate>2017-07-13</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/programming/">programming</a>, <a href="//by-tag/r/">r</a></p></div>  <div>
    <h2>
      <a href="/science/data-science/long-and-wide/" rel="bookmark" title="Permalink to Long and wide matrices">Long and wide matrices</a>
    </h2>
		<div class="summary">A quick lesson on representing data and reshaping matrices</div>
   <p>There are two broad ways data can be stored in a matrix. In "wide" format,
everything for a given subject / record is on a single line / row. For
example:</p>
<div class="highlight"><pre><span></span>patientId   control_cells   control_temp   drugA_cells   drugA_temp   drugB_cells   drugB_temp
1           ...
2           ...
...
</pre></div>


<p>Whereas the "long" format has multiple rows for a subject, one for …</p>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2017-06-11T12:00:00+01:00" pubdate>2017-06-11</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/bioinformatics/">bioinformatics</a>, <a href="//by-tag/academia/">academia</a></p></div>  <div>
    <h2>
      <a href="/science/computational-biology/no-respect/" rel="bookmark" title="Permalink to Why bioinformaticians don't get no respect">Why bioinformaticians don't get no respect</a>
    </h2>
   <p>It's a common cry amongst working bioinformaticians that they're unappreciated, undervalued and generally &quot;get no respect&quot;. While people love to complain, from discussions with peers and colleagues, the same stories come up again and again:</p>
<ul class="simple">
<li>Not being consulted when projects are planned</li>
<li>Technical advice and results not taken seriously</li>
<li>Being …</li></ul>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2017-05-01T12:00:00+01:00" pubdate>2017-05-01</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/programming/">programming</a>, <a href="//by-tag/software/">software</a>, <a href="//by-tag/research/">research</a>, <a href="//by-tag/amazon/">amazon</a>, <a href="//by-tag/aws/">aws</a></p></div>  <div>
    <h2>
      <a href="/science/computational-biology/aws-for-research/" rel="bookmark" title="Permalink to Using AWS for research computing">Using AWS for research computing</a>
    </h2>
		<div class="summary">Your own private computing cluster?</div>
   <p>This is based upon my reply to a question on reddit concerning experiences with using Amazon Web Services (including ELastic Computing, Glacier, etc.) for research.</p>
<p>I was part of a pan-European research consortium that used it for our shared computing infrastructure: databases, a few web apps and web sites, mailers …</p>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2017-02-01T12:00:00+00:00" pubdate>2017-02-01</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/career/">career</a>, <a href="//by-tag/job-search/">job-search</a></p></div>  <div>
    <h2>
      <a href="/science/academia/job-ad-red-flags/" rel="bookmark" title="Permalink to Academic job ad red flags">Academic job ad red flags</a>
    </h2>
		<div class="summary">Words you don't want to read in a job description</div>
   <dl class="docutils">
<dt>&quot;competitive salary&quot;:</dt>
<dd>Like several other terms on this list, this phrase is can be used in a completely sincere manner: <em>We pay decently. If we offer you the job, then we'll negotiate.</em> Unfortunately, it is just as frequently used as a way of avoiding the subject of remuneration, in the …</dd></dl>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2016-09-10T00:00:00+01:00" pubdate>2016-09-10</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/data-science/">data-science</a>, <a href="//by-tag/redcap/">redcap</a>, <a href="//by-tag/databases/">databases</a></p></div>  <div>
    <h2>
      <a href="/science/data-science/more-about-redcap/" rel="bookmark" title="Permalink to Some more things I done learned about REDCap">Some more things I done learned about REDCap</a>
    </h2>
		<div class="summary">A few more surprises</div>
   <p>Refer back to the <a class="reference external" href="/science/data-science/what-i-done-learned-about-redcap">previous article</a> for the background and some introduction to REDCap.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">These notes are based upon REDCap version 6.4.4. Different versions may have fixed or adjusted some of this behaviour.</p>
</div>
<div class="section" id="adding-users">
<h2>Adding users</h2>
<p>To add a user, you need an email address to send a …</p></div>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2016-09-01T00:00:00+01:00" pubdate>2016-09-01</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/academia/">academia</a>, <a href="//by-tag/job-search/">job-search</a>, <a href="//by-tag/interviews/">interviews</a></p></div>  <div>
    <h2>
      <a href="/science/academia/observations-on-job-search/" rel="bookmark" title="Permalink to Random observations on the academic-scientific job search">Random observations on the academic-scientific job search</a>
    </h2>
		<div class="summary">Written without any bitterness at all</div>
   <ul>
<li><p class="first">Maintain employable, valuable skills. Never evince any technical skills. Never casually offer to help a colleague with computer problems, etc. First you'll end up being known for that. Second, people will keep asking you to encrypt their hard-drive, build a database, etc. Third, technical matters are low status &quot;working class …</p></li></ul>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2016-08-15T00:00:00+01:00" pubdate>2016-08-15</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/data-science/">data-science</a>, <a href="//by-tag/databases/">databases</a>, <a href="//by-tag/career/">career</a>, <a href="//by-tag/humour/">humour</a></p></div>  <div>
    <h2>
      <a href="/science/data-science/more-words-not-to-hear/" rel="bookmark" title="Permalink to Words a data scientist never wants to hear">Words a data scientist never wants to hear</a>
    </h2>
		<div class="summary">It never stops.</div>
   <p>Some years ago I wrote a popular piece <a class="reference external" href="/science/computational-biology/words-not-to-hear/">Words a bioinformatician doesn't want to hear</a>. Sadly, a lot of practitioners found it all too real. This time, I'm concentrating on the data end of things.</p>
<hr class="docutils" />
<p>&quot;We really need a database / webapp, so I can query / analyse / visualise this data!&quot;</p>
<p><em>[database …</em></p>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2016-07-01T12:00:00+01:00" pubdate>2016-07-01</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/bayesian/">bayesian</a>, <a href="//by-tag/statistics/">statistics</a>, <a href="//by-tag/likelihood/">likelihood</a></p></div>  <div>
    <h2>
      <a href="/science/data-science/bayesian-stats-in-very-plain-language/" rel="bookmark" title="Permalink to Bayesian stats in very plain language">Bayesian stats in very plain language</a>
    </h2>
		<div class="summary">My pass at explaining the often misunderstood.</div>
   <div class="section" id="introduction">
<h2>Introduction</h2>
<p>Some years ago, I got into an argument with someone abut the relative merits of Bayesian versus Maximum Likelihood in phylogenetics. They asserted the two were basically the same or would come to the same answers. I countered that while they would often agree, they were measuring different things …</p></div>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2015-07-15T00:00:00+01:00" pubdate>2015-07-15</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/data-science/">data-science</a>, <a href="//by-tag/redcap/">redcap</a>, <a href="//by-tag/databases/">databases</a></p></div>  <div>
    <h2>
      <a href="/science/data-science/what-i-done-learned-about-redcap/" rel="bookmark" title="Permalink to What I done learned about REDCap">What I done learned about REDCap</a>
    </h2>
		<div class="summary">A few surprises</div>
   <p>For those not in the know, REDCap is a platform for creating and editing databases through the web. And by and large, it works fine. It saves a lot of development effort. It provides good reporting tools for users. It's secure and robust. But there are some things to be …</p>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2015-05-19T00:00:00+01:00" pubdate>2015-05-19</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/programming/">programming</a>, <a href="//by-tag/science/">science</a>, <a href="//by-tag/r/">r</a>, <a href="//by-tag/computational-biology/">computational-biology</a>, <a href="//by-tag/biomart/">biomart</a>, <a href="//by-tag/genomics/">genomics</a></p></div>  <div>
    <h2>
      <a href="/science/computational-biology/using-biomart/" rel="bookmark" title="Permalink to "Checking gene names & identities in Biomart"">"Checking gene names & identities in Biomart"</a>
    </h2>
   <p>I need to check some gene names and get the sequences for them. Ordinarily, I would prefer to use Python for this sort of thing. Unfortunately, the Python interfaces to Biomart appear to have been written by insane people, so R it is.</p>
<p>What is Biomart? What is <em>a</em> biomart …</p>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2015-05-01T00:00:00+01:00" pubdate>2015-05-01</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/r/">r</a>, <a href="//by-tag/r-studio/">r-studio</a>, <a href="//by-tag/markdown/">markdown</a>, <a href="//by-tag/markup/">markup</a>, <a href="//by-tag/knitr/">knitr</a></p></div>  <div>
    <h2>
      <a href="/science/data-science/markdown-in-rstudio/" rel="bookmark" title="Permalink to Markdown in R Studio">Markdown in R Studio</a>
    </h2>
		<div class="summary">What Markdown markup is allowed.</div>
   <p>If you're doing reproducbility in <a class="reference external" href="http://www.rstuio.com">R Studio</a>, you're probably using knitr. And if you're using <a class="reference external" href="http://yihui.name/knitr/">knitr</a>, you're probably using <a class="reference external" href="https://daringfireball.net/projects/markdown/syntax">Markdown</a>. Unfortunately, due to the lack of a standard for Markdown (and the <a class="reference external" href="http://blog.codinghorror.com/the-future-of-markdown/">subsequent proliferation of various flavours and extensions</a>), it's sometimes not clear what syntax is available to you. Consequently …</p>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2015-03-24T10:10:48+00:00" pubdate>2015-03-24</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/database/">database</a>, <a href="//by-tag/data-science/">data-science</a>, <a href="//by-tag/csv/">csv</a>, <a href="//by-tag/cssql/">cssql</a>, <a href="//by-tag/csvkit/">csvkit</a></p></div>  <div>
    <h2>
      <a href="/science/data-science/rebuilding-databases-with-csv2sql/" rel="bookmark" title="Permalink to (Re-)building databases with csvsql">(Re-)building databases with csvsql</a>
    </h2>
		<div class="summary">Tips and traps when going from a database dump to a database</div>
   <div class="section" id="the-scenario">
<h2>The scenario</h2>
<p>You have a bunch of related CSV files.</p>
<p>Maybe they're the result of a raw database dump. Maybe they've been generated
in some other way: experimental results, various public data sets, whatever. But the important thing is that you need to make a database from them. Perhaps because …</p></div>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2015-02-25T23:29:00+00:00" pubdate>2015-02-25</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/bioinformatics/">bioinformatics</a>, <a href="//by-tag/career/">career</a>, <a href="//by-tag/humour/">humour</a></p></div>  <div>
    <h2>
      <a href="/science/computational-biology/words-not-to-hear/" rel="bookmark" title="Permalink to Words a bioinformatician never wants to hear">Words a bioinformatician never wants to hear</a>
    </h2>
		<div class="summary">Based on hard-won experience.</div>
   <p>(This first appeared on biocodershub.net courtesy of Rad, and has since popped up on coderscrowd.com. It enjoyed some moments of viral popularity, with many aggrieved practitioners chipping in on the comments of the article. Following the resurrection of my website, it's a good opportunity to bring this piece …</p>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2015-02-20T00:00:00+00:00" pubdate>2015-02-20</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/ruby-on-rails/">Ruby-on-Rails</a>, <a href="//by-tag/rails/">Rails</a>, <a href="//by-tag/data-science/">data science</a>, <a href="//by-tag/csv/">CSV</a>, <a href="//by-tag/json/">JSON</a>, <a href="//by-tag/excel/">Excel</a>, <a href="//by-tag/sqlite/">SQLite</a>, <a href="//by-tag/xml/">XML</a>, <a href="//by-tag/redcap/">REDCap</a></p></div>  <div>
    <h2>
      <a href="/science/data-science/tools-for-data/" rel="bookmark" title="Permalink to Tools for data">Tools for data</a>
    </h2>
		<div class="summary">How to store data, what to use</div>
   <p>Prompted by a recent tweet asking what people used for storing and managing their data, I wrote down my own hard-won lessons on the topic. In rough order of preference and data complexity:</p>
<div class="section" id="a-hierarchical-strategy">
<h2>A hierarchical strategy</h2>
<div class="section" id="use-restructured-text-for-documentation">
<h3>Use restructured text for documentation</h3>
<p>Or markdown / asciidoc. The advantages of this being:</p>
<ul class="simple">
<li>It's …</li></ul></div></div>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2013-11-11T17:39:00+00:00" pubdate>2013-11-11</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/academia/">academia</a>, <a href="//by-tag/humour/">humour</a>, <a href="//by-tag/publications/">publications</a></p></div>  <div>
    <h2>
      <a href="/science/academia/philosophical-considerations-in-manuscript-preparation/" rel="bookmark" title="Permalink to Philosophical considerations in manuscript preparation">Philosophical considerations in manuscript preparation</a>
    </h2>
		<div class="summary">Thought experiments and musings.</div>
   <div class="section" id="xeno-s-paradox-of-manuscript-completeness">
<h2>Xeno's paradox of manuscript completeness</h2>
<p>No matter how many drafts you go through, the number of helpful
suggestions make by your co-authors will approach but never quite reach
zero.</p>
</div>
<div class="section" id="plato-s-allegory-of-collaborators-and-the-cave-wall">
<h2>Plato's allegory of collaborators and the cave wall</h2>
<p>Distinguished or influential co-authors have a tendency to invite,
introduce or insist upon …</p></div>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2013-07-11T13:30:00+01:00" pubdate>2013-07-11</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/biopython/">BioPython</a>, <a href="//by-tag/biosequences/">biosequences</a>, <a href="//by-tag/computational-biology/">computational biology</a>, <a href="//by-tag/python/">Python</a></p></div>  <div>
    <h2>
      <a href="/science/computational-biology/biofoo/hitchhikers-guide-to-biopython-sequences-alphabets/" rel="bookmark" title="Permalink to Hitchhikers guide to Biopython: Sequences & alphabets">Hitchhikers guide to Biopython: Sequences &amp; alphabets</a>
    </h2>
   <p>(Originally published on BiocodersHub.)</p>
<p>If you’re doing bioinformatics in Python, you’re probably
using  <a class="reference external" href="http://biopython.org/">Biopython</a>. Actually, Biopython is a
good reason for using Python. But it can be formidable to newcomers:
there’s a lot there and there’s not a huge amount of learning material.
This then is …</p>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2012-09-01T12:00:00+01:00" pubdate>2012-09-01</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/knitr/">knitr</a>, <a href="//by-tag/r/">r</a>, <a href="//by-tag/reproducibility/">reproducibility</a>, <a href="//by-tag/restructured-text/">restructured-text</a></p></div>  <div>
    <h2>
      <a href="/science/data-science/writing-knitr-in-restructured-text/" rel="bookmark" title="Permalink to Writing knitr in restructured text">Writing knitr in restructured text</a>
    </h2>
		<div class="summary">Swapping out Markdown for a different markup.</div>
   <p><em>knitr</em> is a useful R package/tool for documenting analysis. Basically,
it allows the embedding of R code &quot;chunks&quot; within a simple text
document. This document can then be &quot;knitted&quot;, which means that the R
code is interpreted and reinserted in the document along with the
results of that code …</p>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2012-06-14T12:56:00+01:00" pubdate>2012-06-14</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/computational-biology/">computational biology</a>, <a href="//by-tag/galaxy/">galaxy</a></p></div>  <div>
    <h2>
      <a href="/science/computational-biology/galaxy/common-tasks-in-galaxy/" rel="bookmark" title="Permalink to Common tasks in Galaxy">Common tasks in Galaxy</a>
    </h2>
   <p><em>It's all there in the documentation, but sometimes it's hard to find.
This document gives you another place to look.</em></p>
<p>So how do I ...</p>
<div class="section" id="create-admin-users">
<h2>... create admin users?</h2>
<p>Curiously, the identity of admin users is hardcoded into the Galaxy
configuration file. (Which makes it secure, I guess, but separate from
the …</p></div>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2012-06-14T12:56:00+01:00" pubdate>2012-06-14</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
</div>  <div>
    <h2>
      <a href="/science/computational-biology/tools/compiling-quickjoin-and-file-formats/" rel="bookmark" title="Permalink to Compiling Quickjoin and file formats">Compiling Quickjoin and file formats</a>
    </h2>
   <p><em>Problems with building qjoin and getting it to read stockholm files.</em></p>
<p><a class="reference external" href="http://www.birc.dk/Software/QuickJoin">Quickjoin / qjoin</a> is an
excellent commandline program for rapid construction of
neighbour-joining trees. However, while using it recently, I had a few
problems getting it to read Stockholm files, the most accessible of the
formats it can use.</p>
<p>The …</p>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2012-06-14T12:56:00+01:00" pubdate>2012-06-14</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/computational-biology/">computational biology</a>, <a href="//by-tag/galaxy/">galaxy</a></p></div>  <div>
    <h2>
      <a href="/science/computational-biology/galaxy/galaxy-toolsheds/" rel="bookmark" title="Permalink to Galaxy toolsheds">Galaxy toolsheds</a>
    </h2>
   <p>Galaxy toolsheds</p>
<p>Relatively painless tool-sharing</p>
<p>This is a more recent innovation in Galaxy, which can make it a
somewhat confused one: the concept of the toolshed has changed over its
lifetime, the documentation is incomplete, and there's a slightly
strange emphasis in the documentation that exists. So …</p>
<div class="section" id="mile-high-description">
<h2>Mile-high description</h2>
<p>Toolsheds …</p></div>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2012-06-01T12:00:00+01:00" pubdate>2012-06-01</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/computational-biology/">computational-biology</a>, <a href="//by-tag/programming/">programming</a>, <a href="//by-tag/programming-langauges/">programming-langauges</a>, <a href="//by-tag/python/">python</a>, <a href="//by-tag/ruby/">ruby</a></p></div>  <div>
    <h2>
      <a href="/science/computational-biology/language-wars/" rel="bookmark" title="Permalink to Language Wars">Language Wars</a>
    </h2>
		<div class="summary">"What language would you recommend to introduce programming to an audience of life science students at a bachelor level?"</div>
   <p>(Originally published on BiocodersHub)</p>
<p>Following several lengthy and passionate discussions in different venues
on what language to use for teaching bioinformatics, I've started
cutting and pasting my reply. And here it is.</p>
<hr class="docutils" />
<p>You'll get a lot of different opinions on this because:</p>
<ul class="simple">
<li>It's a religious issue. That is, it comes …</li></ul>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2012-02-01T12:00:00+00:00" pubdate>2012-02-01</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/biopython/">BioPython</a>, <a href="//by-tag/biosequences/">biosequences</a>, <a href="//by-tag/computational-biology/">computational biology</a>, <a href="//by-tag/programming/">programming</a>, <a href="//by-tag/python/">Python</a></p></div>  <div>
    <h2>
      <a href="/science/computational-biology/biofoo/hitchhikers-guide-to-biopython-seqrecords/" rel="bookmark" title="Permalink to Hitchhikers guide to BioPython: SeqRecords">Hitchhikers guide to BioPython: SeqRecords</a>
    </h2>
		<div class="summary">For the novice, more-than-raw sequences.</div>
   <p>(Previously published on BiocodersHub.)</p>
<p>Previously I'd spoken about how Biopython represents sequence data with
the Seq class. But there is also the SeqRecord class:</p>
<ul class="simple">
<li>A Seq is just raw sequence data and information about what type of sequence it is.</li>
<li>A SeqRecord is a Seq and all the other information …</li></ul>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2012-01-01T12:00:00+00:00" pubdate>2012-01-01</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
</div>  <div>
    <h2>
      <a href="/science/computational-biology/scripts/cleaning-biosequences/" rel="bookmark" title="Permalink to Cleaning biosequences">Cleaning biosequences</a>
    </h2>
   <p><em>A simple script to check and purge sequence files of possible
problems.</em></p>
<p>Some times you need sequences that are unambiguous (i.e. only 'ACGT',
lacking gaps) whether it's because of the limitations or assumptions of
tools (like omegaMap) or just because you want to know where SNPs or
sequencing ambiguities …</p>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2012-01-01T12:00:00+00:00" pubdate>2012-01-01</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/bioruby/">Bioruby</a>, <a href="//by-tag/phylogenetics/">phylogenetics</a>, <a href="//by-tag/visualisation/">visualisation</a>, <a href="//by-tag/dendroscope/">Dendroscope</a></p></div>  <div>
    <h2>
      <a href="/science/computational-biology/tools/coloring-dendroscope-files/" rel="bookmark" title="Permalink to Coloring dendroscope files">Coloring dendroscope files</a>
    </h2>
		<div class="summary">How to programatically label phylogenies.</div>
   <p>The need had arisen for the tips of a large phylogeny to be labelled in
a systematic way. Rather than &quot;point and click&quot; within Dendroscope, this
script takes a .den/dendro file and colors the tips according to a
&quot;color description&quot; file. This is a simple csv file with taxa …</p>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2012-01-01T12:00:00+00:00" pubdate>2012-01-01</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/bioinformatics/">bioinformatics</a>, <a href="//by-tag/bioruby/">BioRuby</a>, <a href="//by-tag/sequence-analysis/">sequence analysis</a></p></div>  <div>
    <h2>
      <a href="/science/computational-biology/biofoo/consensus-in-bioruby/" rel="bookmark" title="Permalink to Consensus in BioRuby">Consensus in BioRuby</a>
    </h2>
		<div class="summary">Explaining the ill-explained ways to obtain a consensus sequence in BioRuby.</div>
   <p>In BioRuby, alignments are equipped with several methods for obtaining
consensus sequences. Unfortunately, these have terse descriptions which
point you at the BioPerl documentation, with the added bonus of not
<em>quite</em> working like the BioPerl equivalents.</p>
<p>First, let's create a very simple alignment, where everything agrees
except the last sequence …</p>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2012-01-01T12:00:00+00:00" pubdate>2012-01-01</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/computational-biology/">computational biology</a>, <a href="//by-tag/galaxy/">galaxy</a></p></div>  <div>
    <h2>
      <a href="/science/computational-biology/galaxy/galaxy-miscellanea/" rel="bookmark" title="Permalink to Galaxy miscellanea">Galaxy miscellanea</a>
    </h2>
   <p><em>Odds and ends and the surprising.</em></p>
<div class="section" id="redirects">
<h2>Redirects</h2>
<p>If you are serving the installation with a proxy redirect (e.g. the
galaxy server is running on port 7070 but is being redirect by Apache to
appear at port 80 on <em>/galaxy</em>), while you can access Galaxy at both
addresses, login will …</p></div>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2012-01-01T12:00:00+00:00" pubdate>2012-01-01</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
</div>  <div>
    <h2>
      <a href="/science/computational-biology/tools/more-about-mrbayes/" rel="bookmark" title="Permalink to More about MrBayes">More about MrBayes</a>
    </h2>
   <p>Some (more) notes about the venerable Bayesian reconstruction program.</p>
<div class="section" id="error-when-setting-parameter-gap-2">
<h2>Error when setting parameter &quot;Gap&quot; (2)</h2>
<p>When attempting to execute a Nexus file, MrBayes kept spitting back this
cryptic error upon loading:</p>
<pre class="literal-block">
Executing file &quot;c_vp1_nuc_seqs.nxs&quot; [...]
Reading data block Allocated matrix [...]
Data is Dna Gap character matches matching or missing characters …</pre></div>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2012-01-01T12:00:00+00:00" pubdate>2012-01-01</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/computational-biology/">computational biology</a>, <a href="//by-tag/galaxy/">galaxy</a></p></div>  <div>
    <h2>
      <a href="/science/computational-biology/galaxy/more-things-i-done-learned-about-galaxy-tool-development/" rel="bookmark" title="Permalink to More things I done learned about Galaxy tool development">More things I done learned about Galaxy tool development</a>
    </h2>
   <p><em>More things I done learned about Galaxy tool development</em></p>
<p>A pot-pouri of titbits that are probably documented somewhere, but
weren't obvious to me.</p>
<div class="section" id="labels-on-the-tool-menu">
<h2>Labels on the tool menu</h2>
<p>You can't nest sections on the tool panel - but you can put labels in
sections for the same effect. Look at tool_conf …</p></div>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2012-01-01T12:00:00+00:00" pubdate>2012-01-01</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
</div>  <div>
    <h2>
      <a href="/science/academia/ross-crozier-1943-2009/" rel="bookmark" title="Permalink to Ross Crozier 1943-2009">Ross Crozier 1943-2009</a>
    </h2>
   <p>The sudden death of Ross Crozier on the 12th of November was heralded
largely by a slow ripple of email, phone calls and Facebook messages
across the globe.  I found out from an email that started with a short
but singularly complete sentence:</p>
<blockquote>
<em>Terrible news.</em></blockquote>
<p>It is sobering to think …</p>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2012-01-01T12:00:00+00:00" pubdate>2012-01-01</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/science/">science</a>, <a href="//by-tag/bioinformatics/">bioinformatics</a>, <a href="//by-tag/ngs/">ngs</a></p></div>  <div>
    <h2>
      <a href="/science/computational-biology/what-works-ngs-assemblers/" rel="bookmark" title="Permalink to What works - NGS assemblers">What works - NGS assemblers</a>
    </h2>
		<div class="summary">A quick paper review on picking the best assembler</div>
   <p>You could spend all day just keeping up with developments in
next-generation sequencing. Companies announce new and revolutionary
technologies seemingly every month, promising to do more, better and for
less. Yet at the same time, it’s difficult to hack your way through the
marketing tallk and get hard figures …</p>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2011-01-01T12:00:00+00:00" pubdate>2011-01-01</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/ruby/">ruby</a>, <a href="//by-tag/bioinformatics/">bioinformatics</a>, <a href="//by-tag/sequence-analysis/">sequence-analysis</a></p></div>  <div>
    <h2>
      <a href="/science/computational-biology/scripts/drawing-sequence-logos/" rel="bookmark" title="Permalink to Drawing sequence logos">Drawing sequence logos</a>
    </h2>
		<div class="summary">A very simple script to do a simple but tedious task.</div>
   <p>Sequence logos are a common way of representing SNPs and diversity in
groups of sequences. This script automates the task. It's a bit rough
around the edges and serves mainly as a base for further hacking.</p>
<p>Usage is:</p>
<pre class="literal-block">
drawlogo.rb [options] FILE1 [FILE2 ...]
</pre>
<p>where options are:</p>
<blockquote>
<table class="docutils option-list" frame="void" rules="none">
<col class="option" />
<col class="description" />
<tbody valign="top">
<tr><td class="option-group">
<kbd><span class="option">-h</span>, <span class="option">--help</span></kbd></td>
<td>Display this …</td></tr></tbody></table></blockquote>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2010-06-01T12:00:00+01:00" pubdate>2010-06-01</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/imported/">imported</a>, <a href="//by-tag/sequence-analysis/">sequence-analysis</a>, <a href="//by-tag/script/">script</a>, <a href="//by-tag/ruby/">ruby</a>, <a href="//by-tag/bioruby/">bioruby</a>, <a href="//by-tag/bioinformatics/">bioinformatics</a></p>     <div class="attention"><b>Attention:</b> This article has been imported from a previous website and has not yet been checked. It may be malformed or incomplete.
     </div>
</div>  <div>
    <h2>
      <a href="/science/computational-biology/scripts/reducing-a-sequence-to-snps/" rel="bookmark" title="Permalink to Reducing a sequence to SNPs">Reducing a sequence to SNPs</a>
    </h2>
		<div class="summary">A script for a simple task.</div>
   <p>A largely self-explanatory script. This will &quot;shrink&quot; an alignment,
deleting all sites that don't contain a polymorphism in some member
sequence. A little bit of script candy as well, this takes any number of
files and saves the results in a new file named according to a definable
schema:</p>
<pre class="literal-block">
#!/usr …</pre>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2010-03-01T12:00:00+00:00" pubdate>2010-03-01</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/science/">science</a>, <a href="//by-tag/tools/">tools</a>, <a href="//by-tag/phylogenetics/">phylogenetics</a>, <a href="//by-tag/web-development/">web-development</a></p></div>  <div>
    <h2>
      <a href="/science/computational-biology/jsphylosvg/" rel="bookmark" title="Permalink to jsPhyloSVG">jsPhyloSVG</a>
    </h2>
   <p><a class="reference external" href="http://jsphylosvg.com/">jsPhyloSVG</a> (or JPS from this point) is a
nifty Javascript library for displaying phylogenetic trees in your
browser. It can:</p>
<ul class="simple">
<li>accept trees in a number of formats, encoded in the page or on file or loaded via Ajax</li>
<li>allow these trees to be exported or saved as SVG</li>
<li>display trees …</li></ul>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2009-06-01T12:00:00+01:00" pubdate>2009-06-01</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/osgb/">osgb</a></p></div>  <div>
    <h2>
      <a href="/science/geospatial/ordnance-survey-locations/" rel="bookmark" title="Permalink to Ordnance Survey locations">Ordnance Survey locations</a>
    </h2>
		<div class="summary">Converting between OS grid references and longitudes-latitudes.</div>
   <p>The Ordinance Survey is a UK-peculiar geospatial format, ubiquitous via
street atlases, hiking charts and (yes) farming and epidemiological
maps. It is explained in great detail is several places, but here's a
quick overview:</p>
<p>The OS grid is a set of 25 squares, 500 kilometers a side, arranged
5-by-5 that …</p>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2009-01-01T12:00:00+00:00" pubdate>2009-01-01</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/mr-bayes/">mr-bayes</a>, <a href="//by-tag/phylogenetics/">phylogenetics</a>, <a href="//by-tag/possibly-obselete/">possibly-obselete</a></p>     <div class="attention"><P><b>Attention:</b> This article may refer to information that is outdated or no longer relevant. It is left here for historical purposes.</P></div>
</div>  <div>
    <h2>
      <a href="/science/computational-biology/tools/error-1-for-mrbayes/" rel="bookmark" title="Permalink to Error 1 for Mrbayes">Error 1 for Mrbayes</a>
    </h2>
		<div class="summary">What happens when a make fails.</div>
   <p>If this happens to you when trying to compile MrBayes:</p>
<pre class="literal-block">
% make
gcc -DUNIX_VERSION -DUSE_READLINE -O3 -Wall    -c -o mb.o mb.c
gcc -DUNIX_VERSION -DUSE_READLINE -O3 -Wall    -c -o mcmc.o mcmc.c
gcc -DUNIX_VERSION -DUSE_READLINE -O3 -Wall    -c -o bayes.o bayes.c
bayes.c:45:31: readline/readline …</pre>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2007-06-14T12:56:00+01:00" pubdate>2007-06-14</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/bioruby/">bioruby</a>, <a href="//by-tag/sequences/">sequences</a>, <a href="//by-tag/database/">database</a></p></div>  <div>
    <h2>
      <a href="/science/computational-biology/scripts/fetch-sequences-from-db/" rel="bookmark" title="Permalink to Fetch sequences from db">Fetch sequences from db</a>
    </h2>
		<div class="summary">A simple script to grab bioseqs by accession.</div>
   <p>This just wraps the BioRuby fetch functionality in a friendly
commandline interface. In brief, it can accept accession ids on the
commandline or from a piped file (one accession per line) and save the
corresponding sequences from the db. Sequences may be downloaded via the
bioruby or EBI servers. The …</p>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2007-06-14T12:56:00+01:00" pubdate>2007-06-14</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/computational-biology/">computational biology</a>, <a href="//by-tag/galaxy/">galaxy</a></p></div>  <div>
    <h2>
      <a href="/science/computational-biology/galaxy/installing-galaxy/" rel="bookmark" title="Permalink to Installing Galaxy">Installing Galaxy</a>
    </h2>
		<div class="summary">Setting up a production version of GMOD Galaxy for general use.</div>
   <p>This presents one way to create an optimized production Galaxy instance.
Variations are certainly possible and some of the choices presented
are/were dictated by local culture. Certain settings may be more
suitable for production or development environments. Nonetheless, this
presents a start-to-stop process for installation and setup.</p>
<p><strong>Note:</strong> this …</p>
  </div>
</article><article class="summary">
<div class="metadata">
  <time datetime="2007-01-01T12:00:00+00:00" pubdate>2007-01-01</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
</div>  <div>
    <h2>
      <a href="/science/computational-biology/tools/parsing-dendroscope-nodes/" rel="bookmark" title="Permalink to Parsing Dendroscope nodes">Parsing Dendroscope nodes</a>
    </h2>
   <p>For when you have to do lots to a big tree.</p>
<p>Previously, I showed how Dendroscope files can be easily manipulated
with brute-force regex, so you can right scripts to color a mass of
nodes, rather than having to format them one-by-one in the GUI. However,
more complex manipulations require …</p>
  </div>
</article>

		  </div>

		  <div class="sidebar">
		    <div class="sidebar-container" >


	            <aside>
	              <h2>About</h2>
			      <p class='summary'>
                  This site has just been restored, with most of the content
						translated automatically from its old form. Expect bugs,
						missing information and general wierdness as things
						get sorted out.
			      </p>
			    </aside>


  	          <nav>
	            <h2>Categories</h2>
	            <ul>
	                <li ><a href="/misc/">misc</a></li>
	                <li ><a href="/personal/">personal</a></li>
	                <li ><a href="/programming/">programming</a></li>
	                <li ><a href="/publications/">publications</a></li>
	                <li class="active"><a href="/science/">science</a></li>
	                <li ><a href="/software/">software</a></li>
	                <li ><a href="/talks/">talks</a></li>
	            </ul>

	            <ul>
	              <li><a href="/tags.html">all tags</a></li>
	            </ul>
	          </nav>

	            <aside>
	            <h2>Social</h2>
			      <ul class="social">
				    <li><a href="http://twitter.com/agapow">Twitter</a><i></i></li>
				    <li><a href="http://github.com/agapow">Github</a><i></i></li>
				    <li><a href="https://plus.google.com/u/0/113920832845886210876/posts">Google+</a><i></i></li>
				    <li><a href="http://uk.linkedin.com/pub/paul-michael-agapow/2/15b/184/">linkedIn</a><i></i></li>
			      </ul>
			    </aside>

	            <aside>
	              <h2>Links</h2>
	              <ul>
	                  <li><a href="http://www.imperial.ac.uk/data-science/">IC Data Science Institute</a></li>
	                  <li><a href="http://www.meetup.com/Bioinformatics-London">Bioinformatics London</a></li>
	              </ul>
	            </aside>

<a class="twitter-timeline" data-dnt="true" href="https://twitter.com/agapow" data-widget-id="336428520026812416">Tweets by @agapow</a>
<script>!function(d,s,id){var js,fjs=d.getElementsByTagName(s)[0],p=/^http:/.test(d.location)?'http':'https';if(!d.getElementById(id)){js=d.createElement(s);js.id=id;js.src=p+"://platform.twitter.com/widgets.js";fjs.parentNode.insertBefore(js,fjs);}}(document,"script","twitter-wjs");</script>

	        </div>
		  </div>

	  </div>

      <footer>
		<p role="contentinfo">
		  Paul Agapow - Proudly powered by <a href="http://alexis.notmyidea.org/pelican/">pelican</a>. Theme <a href="https://github.com/fle/pelican-sober">pelican-sober</a>.
    	</p>

	  </footer>

	</div>


</body>
</html>