<!doctype html>
<html lang="">
<head>
	<meta charset="utf-8"/>
	<title>What works - NGS assemblers - Make More Machines</title>
	<meta name="author" content="Paul Agapow">


  <meta name="description" content="You could spend all day just keeping up with developments in next-generation sequencing. Companies announce new and revolutionary technologies seemingly every month, promising to do more, better and for less. Yet at the same time, it’s difficult to hack your way through the marketing tallk and get hard figures …">

	<meta name="twitter:card" content="summary">
	  <meta name="twitter:creator" content="@agapow">
	<meta name="twitter:title" content="What works - NGS assemblers">
	<meta name="twitter:description" content="You could spend all day just keeping up with developments in next-generation sequencing. Companies announce new and revolutionary technologies seemingly every month, promising to do more, better and for less. Yet at the same time, it’s difficult to hack your way through the marketing tallk and get hard figures …">
	<meta name="twitter:url" content="/science/computational-biology/what-works-ngs-assemblers/">


	<link rel="stylesheet" href="/theme/css/main.css" type="text/css" />



</head>

<body>

    <div class="container">

	  <header role="banner">
	    <div class="feeds">
	    </div>
	      <nav class="pages">
			  <a href="/about">About</a>
-			  <a href="/whereabouts">Whereabouts</a>
	      </nav>
		<a href="" class="title">Make More Machines</a>&nbsp;&nbsp;(agapow.net)
      </header>

	  <div class="wrapper">

		  <div role="main" class="content">
	<article class="full">

		<h1>What works - NGS assemblers</h1>

<div class="metadata">
  <time datetime="2012-01-01T12:00:00+00:00" pubdate>2012-01-01</time>
    <address class="vcard author">
      by <a class="url fn" href="/False">Paul Agapow</a>
    </address>
  in <a href="/science/">science</a>
<p class="tags">tagged <a href="//by-tag/science/">science</a>, <a href="//by-tag/bioinformatics/">bioinformatics</a>, <a href="//by-tag/ngs/">ngs</a></p></div>
		<div class="summary">A quick paper review on picking the best assembler</div>


      <p>You could spend all day just keeping up with developments in
next-generation sequencing. Companies announce new and revolutionary
technologies seemingly every month, promising to do more, better and for
less. Yet at the same time, it’s difficult to hack your way through the
marketing tallk and get hard figures. And the road of NGS technologies
is strewn with once promising technologies …</p>
<p>Fortunate then that a few recent papers have looked at NGS tech and
done some hard comparative studies. For example:</p>
<blockquote>
Finotello et al. (2012)&nbsp;Comparative analysis of algorithms for
whole-genome assembly of pyrosequencing data. <em>Brief
Bioinform</em>&nbsp;13(3):269-280.&nbsp;doi:&nbsp;10.1093/bib/bbr063</blockquote>
<p>Here the vital step we’re talking about is assemblers: how accurate is
the assembly, where and how many gaps result, how good are they at
solving complex genetic regions.</p>
<p>Of course, the test data set is perhaps critical context: what works
well for one dataset may be inappropriate to others. Here it’s 454 reads
of three different bacterial genomes for which high quality reference
genomes were available: &nbsp;<em>Zymomonas mobilis</em>, <em>Helicobacter
pylori</em>&nbsp;and&nbsp;<em>E. coli.</em>&nbsp;Furthermore, the effects of different degrees of
coverage were estimated by sampling the original data to different
degrees. So the study is essentially doing de novo assembly with
different software and then assessing the results by comparison against
a reference genome.</p>
<p>The assemblers in question are:</p>
<ul class="simple">
<li>Newbler (version 2.3 and 2.5)</li>
<li>CABOG</li>
<li>A pre-release version of PCAP</li>
<li>MIRA</li>
<li>CLC Assembly Cell</li>
</ul>
<p>(Note: this seems like a slightly odd selection of assemblers, but it
may be just my personal experience speaking). A number of popular
assemblers were excluded:</p>
<ul class="simple">
<li>Long read assemblers like PHRAP and PCAP were not used due to
previous work showing them were suboptimal</li>
<li>Short read assemblers like Velvet and ABySS were not used due to
previous work showing they produce a lot of small contigs and poorer
reconstruction</li>
</ul>
<p>So what are the results? There’s a mass of figures and graphs in there
that are&nbsp;difficult&nbsp;to digest but to brutally sumarize them:</p>
<ul class="simple">
<li>CABOG gives the largest and most complete contiguous assemblies.
Emphasis is on the word “contiguous” – CABOG is very good at putting
things together, although it sometimes does this incorrectly</li>
<li>Newbler creates very accurate contigs, although more “disconnected”.
There’s a curious story here where the more recent version of the
software (v2.5) is better at linking the contigs at the expense of
more errors</li>
<li>CLC copes better with lower coverage than other assemblers</li>
<li>MIRA has a significant error rate is assembling contigs</li>
<li>PCAP shows a somewhat erratic performance (perhaps because of being
pre-release)</li>
</ul>
<p>As an extra, the authors also ask what the “sweet spot” for coverage
is – how much is “enough”, how much effort should you invest in
increasing coverage? All assemblers show a rapid improvement up to about
15-20x coverage and then almost none as far as 72x coverage. (Note this
is much lower than previously suggested figures.)&nbsp;The authors take-home
is that more than 20-30x coverage is unlikely to bring much benefit.&nbsp;In
fact they suggest that exceeding this will not only be expensive but may
compromise assembly quality. (The reasoning for this last point is
unclear to me – perhaps the additional of more but uninformative data
will just create &nbsp;larger job for the assembler.)</p>
<p>The results are a little surprising: no one in my circles uses CABOG
or it’s predecessors, with Mira has a reasonable following. And some of
the differences are quite substantial, on the level of several percent
of the genome being “missed”. Which makes me wonder how much NGS data
we’re going to have to throw out in the coming years when we find how
bad it is …</p>


	</article>

    <p>
	<a href="https://twitter.com/share" class="twitter-share-button" data-via="" data-lang="en" data-size="large" data-related="">Tweet</a>
	<script>!function(d,s,id){var js,fjs=d.getElementsByTagName(s)[0];if(!d.getElementById(id)){js=d.createElement(s);js.id=id;js.src="//platform.twitter.com/widgets.js";fjs.parentNode.insertBefore(js,fjs);}}(document,"script","twitter-wjs");</script>
	</p>

<div id="disqus_thread"></div>
<script type="text/javascript">
    /* * * CONFIGURATION VARIABLES * * */
    var disqus_shortname = 'agapow';

    /* * * DON'T EDIT BELOW THIS LINE * * */
    (function() {
        var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
        dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
        (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
    })();
</script>
<noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript" rel="nofollow">comments powered by Disqus.</a></noscript>

		  </div>

		  <div class="sidebar">
		    <div class="sidebar-container" >


	            <aside>
	              <h2>About</h2>
			      <p class='summary'>
                  This site has just been restored, with most of the content
						translated automatically from its old form. Expect bugs,
						missing information and general wierdness as things
						get sorted out.
			      </p>
			    </aside>


  	          <nav>
	            <h2>Categories</h2>
	            <ul>
	                <li ><a href="/misc/">misc</a></li>
	                <li ><a href="/personal/">personal</a></li>
	                <li ><a href="/programming/">programming</a></li>
	                <li ><a href="/publications/">publications</a></li>
	                <li class="active"><a href="/science/">science</a></li>
	                <li ><a href="/software/">software</a></li>
	                <li ><a href="/talks/">talks</a></li>
	            </ul>

	            <ul>
	              <li><a href="/tags.html">all tags</a></li>
	            </ul>
	          </nav>

	            <aside>
	            <h2>Social</h2>
			      <ul class="social">
				    <li><a href="http://twitter.com/agapow">Twitter</a><i></i></li>
				    <li><a href="http://github.com/agapow">Github</a><i></i></li>
				    <li><a href="https://plus.google.com/u/0/113920832845886210876/posts">Google+</a><i></i></li>
				    <li><a href="http://uk.linkedin.com/pub/paul-michael-agapow/2/15b/184/">linkedIn</a><i></i></li>
			      </ul>
			    </aside>

	            <aside>
	              <h2>Links</h2>
	              <ul>
	                  <li><a href="http://www.imperial.ac.uk/data-science/">IC Data Science Institute</a></li>
	                  <li><a href="http://www.meetup.com/Bioinformatics-London">Bioinformatics London</a></li>
	              </ul>
	            </aside>

<a class="twitter-timeline" data-dnt="true" href="https://twitter.com/agapow" data-widget-id="336428520026812416">Tweets by @agapow</a>
<script>!function(d,s,id){var js,fjs=d.getElementsByTagName(s)[0],p=/^http:/.test(d.location)?'http':'https';if(!d.getElementById(id)){js=d.createElement(s);js.id=id;js.src=p+"://platform.twitter.com/widgets.js";fjs.parentNode.insertBefore(js,fjs);}}(document,"script","twitter-wjs");</script>

	        </div>
		  </div>

	  </div>

      <footer>
		<p role="contentinfo">
		  Paul Agapow - Proudly powered by <a href="http://alexis.notmyidea.org/pelican/">pelican</a>. Theme <a href="https://github.com/fle/pelican-sober">pelican-sober</a>.
    	</p>

	  </footer>

	</div>


</body>
</html>